### ISSUE TITLE
[FEA] Support datetime dtype columns in `Series.apply`

### ISSUE THREAD

**Issue by miguelusque** [rapidsai/cudf#938]
**Describe the bug**
When invoking _applymap_ on a Datetime column, I get the following error. Am I doing something wrong? Is there any workaround?

I have tried the _applymap_ on the same DF but different column type (*int*), and it seems to work.

**Steps/Code to reproduce bug**
<img width="897" alt="screenshot 2019-02-13 at 23 10 19" src="https://user-images.githubusercontent.com/26169771/52748099-e5260080-2fe5-11e9-9f3b-496dca08f6b0.png">

**Expected behavior**
No error.

**Environment details (please complete the following information):**
Tested on DGX-1. RAPIDS 0.5.1,  container cuda 10, runtime version.
Also tested on DGX-1. cudf 0.6.0,  container cuda 10, devel version + git clone cudf 0.6 branch.

**Additional context**

**Comment by miguelusque** [comment]
Hi @kkraus14 ,

Thank you for your reply.

The issue I was trying to highlight is that _applymap_ does not support `datetime` series.

When I run the following code:

```
import cudf

dates = cudf.DataFrame({'a':['2021-01-01', '2021-01-02', '2031-01-03']})
dates["a"] = dates["a"].astype('datetime64[s]')

def nop(date_column):
    return date_column
    
result = dates["a"].applymap(nop)
```

I get the this error:
`AttributeError: 'DatetimeColumn' object has no attribute 'unary_operator'`

Maybe, this issue might be changed from 'BUG' to 'FEA' request if this is a known limitation.

Regards,
Miguel

**Comment by vyasr** [comment]
`applymap` is deprecated, so future support around these features should be considered in the context of `apply` instead.

**Comment by miguelusque** [comment]
Hi @vyasr , I agree that `applymap` method is deprecated, but if it can be considered for the `apply` method, this issue should remain opened until fixed.

**Comment by vyasr** [comment]
@miguelusque fair enough. I've updated the title. At present calling `apply` on a column with an unsupported dtype should give a comprehensible error, so I've recategorized this as a feature request.

**Comment by vyasr** [comment]
@beckernick JFYI there's also #6756 which I just realized was a superset of this issue. We may want to split that one up into separate pieces and close it (with this issue being one of those pieces).

**Comment by miguelusque** [comment]
> The no-op example in this thread now works. @miguelusque , could you share an example of the kind of UDF you'd like to be able to use with `.apply`?

Hi @beckernick , being 100% honest, I do not recall which operation I needed to perform at that moment. I was porting some customer code to cuDF at that moment, but I do not remember the details. Sorry.

But I do remember that I did find a workaround. :-)

Thank you for considering this FR 

Hi @vyasr , I am happy with closing this FR and keeping track of it in #6756. Please, do not hesitate to do it if you think that it is the way to go. Thanks!

Miguel

**Comment by vyasr** [comment]
To close the loop, the below snippet now works in cudf, so I'm going to close this issue as resolved and we can track anything else in #6756:
```
In [1]: import cudf
   ...: 
   ...: dates = cudf.DataFrame({'a':['2021-01-01', '2021-01-02', '2031-01-03']})
   ...: dates["a"] = dates["a"].astype('datetime64[s]')
   ...: 
   ...: def nop(date_column):
   ...:     return date_column
   ...: 
   ...: result = dates["a"].apply(nop)
   ...: 

In [2]: result
Out[2]: 
0   2021-01-01
1   2021-01-02
2   2031-01-03
Name: a, dtype: datetime64[s]
```
